src/WSRequestHandler_Sources.cpp

```diff
@@ -187,33 +193,45 @@ RpcResponse WSRequestHandler::GetVolume(const RpcRequest& request)
 
 	OBSDataAutoRelease response = obs_data_create();
 	obs_data_set_string(response, "name", obs_source_get_name(source));
-	obs_data_set_double(response, "volume", obs_source_get_volume(source));
+	if (!useDecibel) {
+		obs_data_set_double(response, "volume", obs_source_get_volume(source));
+	} else {
+		float volume = obs_source_get_volume(source);
+		obs_data_set_double(response, "volume", obs_mul_to_db(volume));
+	}
 	obs_data_set_bool(response, "muted", obs_source_muted(source));
 
 	return request.success(response);
 }
 
 /**
- * Set the volume of the specified source.
- *
- * @param {String} `source` Source name.
- * @param {double} `volume` Desired volume. Must be between `0.0` and `1.0`.
- *
- * @api requests
- * @name SetVolume
- * @category sources
- * @since 4.0.0
- */
+* Set the volume of the specified source. Default request format uses mul, NOT SLIDER PERCENTAGE.
+*
+* @param {String} `source` Source name.
+* @param {double} `volume` Desired volume. Must be between `0.0` and `1.0` for mul, and under 0.0 for dB. Note: OBS will interpret dB values under -100.0 as Inf.
+* @param {boolean (optional)} `useDecibel` Interperet `volume` data as decibels instead of amplitude/mul.
+*
+* @api requests
+* @name SetVolume
+* @category sources
+* @since 4.0.0
+*/
 RpcResponse WSRequestHandler::SetVolume(const RpcRequest& request)
  {
 	if (!request.hasField("source") || !request.hasField("volume")) {
 		return request.failed("missing request parameters");
 	}
 
+	bool useDecibel = false;
+	if (request.hasField("useDecibel")) {
+		useDecibel = obs_data_get_bool(request.parameters(), "useDecibel");
```

Comment:
The return value of `obs_data_get_bool` defaults to false if the value can't be found, so you can simplify this to:

```cpp
bool useDecibel = obs_data_get_bool(request.parameters(), "useDecibel");
```

without the `if` statement